File: day1/Python/part2.py
import re
with open("input.txt", "r") as file:
    contents = file.read()

arr = re.split(r'\r?\n',contents)
count = 0

for i in range(3,len(arr)):
    if (int(arr[i]) + int(arr[i-1]) + int(arr[i-2])) > (int(arr[i-1]) + int(arr[i-2]) + int(arr[i-3])):
         count+=1

print(count)

File: day1/JS/part2.js
const fs = require('fs');
(function(){
    let input = fs.readFileSync('input.txt','utf-8').split(/\r?\n/);
    const arr = Array.from(input);

    let newArr = [];

    for(let i=3; i < arr.length; i++){
        const sum = parseInt(arr[i])+ parseInt(arr[i-1]) + parseInt(arr[i-2]);
        newArr.push(sum);
    }

    function compare(element,index,array){
        if (index == 0){
            return true;
        }
        if (element > array[index-1]){
            return true;
        }
    }

    console.log(newArr.filter(compare).length);
})();